File: src/main/java/com/sisqueslabs/nexora/api/contexts/chat/application/port/JobsGateway.java
package com.sisqueslabs.nexora.api.contexts.chat.application.port;

import com.sisqueslabs.nexora.api.contexts.chat.domain.entities.Job;

/**
 * Manages the job lifecycle in nexora-jobs.
 */
public interface JobsGateway {

    Job create(String model);

    Job markScheduled(String jobId, String nodeId);

    Job markRunning(String jobId);

    Job markCompleted(String jobId);

    Job markFailed(String jobId, Throwable cause);
}

File: src/contexts/chat/application/ports/jobs.gateway.ts
import { JobAggregate } from '@contexts/chat/domain/aggregates/job.aggregate';

export const JOBS_GATEWAY = Symbol('JOBS_GATEWAY');

/**
 * Manages the job lifecycle in nexora-jobs.
 */
export interface IJobsGateway {
  create(model: string): Promise<JobAggregate>;
  markScheduled(jobId: string, nodeId: string): Promise<JobAggregate>;
  markRunning(jobId: string): Promise<JobAggregate>;
  markCompleted(jobId: string): Promise<JobAggregate>;
  markFailed(jobId: string, cause: unknown): Promise<JobAggregate>;
}
